File: P1.m
try
  
pkg load database
conn = pq_connect(setdbopts('dbname','postgres','host','localhost','port','5432','user','postgres','password','2807289050109'))

disp("1. Jugar")
disp("2. Historial")
main=input("Opción: ");
switch main
 case 1
 dado1=randi(6)
 dado2=randi(6)
 disp("El valor del dado 1 es: ")
 disp(dado1)
 disp("El valor del dado 2 es: ")
 disp(dado2)
 suma = dado1+dado2;
 disp("La suma de los dados es: ")
 disp(suma)
 if suma == 7
   disp("Perdiste :/")
   res = ('Perdiste'); 
   pq_exec_params(conn, 'insert into programa1 values ($1,$2,$3,$4);',{dado1,dado2,suma,res});
 elseif  suma == 8
   disp("Ganaste :D")
   res = ('Ganaste');
   pq_exec_params(conn, 'insert into programa1 values ($1,$2,$3,$4);',{dado1,dado2,suma,res});
 else
   disp("Juega de nuevo")
   res = ('Juega de nuevo');
   pq_exec_params(conn, 'insert into programa1 values ($1,$2,$3,$4);',{dado1,dado2,suma,res});
 endif
 case 2
 pq_exec_params (conn, "select * from programa1;")
 otherwise
 disp("Escoge una opción válida")
end
catch
 disp("Algo salió mal")
end_try_catch

File: P1.py
import random
import psycopg2

try: 
    conexion = psycopg2.connect(
        host = "localhost",
        port = "5432",
        user = "postgres",
        password = "2807289050109",
        dbname = "postgres"
    )
    print("Conexión exitosa")
except psycopg2.Error as e:
    print("Ocurrió un error en la conexión")
    print("Verifique los parámetros")

def main():

    print ("\nMenú ")
    print ("\n1.Jugar")
    print ("2.Historial")

    opcion = input("\nEscoge la opción deseada: ")
    cursor = conexion.cursor()

    if opcion == "1":

        dado1 = random.randint(1, 6)
        dado2 = random.randint(1, 6)
        print('El valor del dado 1 es: ',dado1)
        print('El valor del dado 2 es: ',dado2)

        suma = dado1 + dado2
        print("La suma de los dados es: ", suma)

        if suma == 7 :
            print("Perdiste :/")
            res = "Perdiste"
            cursor.execute("insert into programa1(dado1, dado2, suma, resultado) values(%s, %s,%s,%s);",(dado1,dado2,suma,res))
            conexion.commit()
            cursor.close()
            conexion.close()

        elif suma == 8 :
            print("Ganaste :D")
            res= "Ganaste"
            cursor.execute("insert into programa1(dado1, dado2, suma, resultado) values(%s, %s,%s,%s);",(dado1,dado2,suma,res))
            conexion.commit()
            cursor.close()
            conexion.close()
        
        else:
            print("Juega de nuevo")
            res = "Juega de nuevo"
            cursor.execute("insert into programa1(dado1, dado2, suma, resultado) values(%s, %s,%s,%s);",(dado1,dado2,suma,res))
            conexion.commit()
            cursor.close()
            conexion.close()

    if opcion == "2":

        cursor = conexion.cursor()
        SQL = 'select * from programa1;'
        cursor.execute(SQL)
        valores = cursor.fetchall()
        print(valores)

main()
    